MapboxNavigation/NavigationViewController.swift

```diff
@@ -651,8 +655,13 @@ extension NavigationViewController: RouteControllerDelegate {
 
 extension NavigationViewController: TunnelIntersectionManagerDelegate {
     public func tunnelIntersectionManager(_ manager: TunnelIntersectionManager, willEnableAnimationAt location: CLLocation) {
-        self.routeController?.tunnelIntersectionManager(manager, willEnableAnimationAt: location)
-        self.styleManager.applyStyle(type: .night)
+        guard let routeController else {
+            return
+        }
+        routeController.tunnelIntersectionManager(manager, willEnableAnimationAt: location)
+        // If we're in a tunnel at sunrise, don't let the timeOfDay timer clobber night mode
+        self.styleManager.cancelTimeOfDayTimer()
```

Comment:
Good catch to cancel this when one is in a tunnel. But don't we need to re-add the timer once the user leaves the tunnel?

Reply:
Fair question! I should have called this out at the top level, but this is actually not a change in behavior. Previously this logic lived in `applyStyle`. 

I was working on applyStyle and could not figure out why we'd cancel the the timer. Eventually I realized it was only for this case, so I extracted the functionality here.

After you exit the tunnel, same as before, `self.styleManager.timeOfDayChanged()` is called which restarts the timer.


e.g. the green area roughly corresponds to a tunnel over the highway:

https://github.com/user-attachments/assets/236d7081-1462-477f-a3e3-0c02e77f433e

Reply:
A little more context in: https://github.com/maplibre/maplibre-navigation-ios/pull/65/commits/f8eb874453750c16a639ba3c0614df9fd19abd6d